py/src/braintrust/logger.py

```diff
@@ -197,14 +201,33 @@ def flush(self, initial_items=None, batch_size=100):
                     break
 
             if len(items) > 0:
-                conn.post_json("logs", items)
+                # Construct batches of items that do not exceed the max request size divided by 2
```

Comment:
I am implementing exactly the same thing in the AWS lambda publisher. My implementation looks like this:

```
    def flush(self, initial_items=None, batch_size=100):
        initial_items = initial_items or []
        initial_items.reverse()
        items_len = 0
        while True:
            while len(items) < batch_size and items_len < MAX_PUBLISH_SIZE / 2:
                if len(initial_items) > 0:
                    items.append(initial_items.pop())
                else:
                    try:
                        items.append(self.queue.get_nowait())
                    except queue.Empty:
                        break
                items_len += len(json.dumps(items[-1]))
            if len(items) == 0:
                break
        # Do publish
```

While true that we will exceed `MAX_PUBLISH_SIZE / 2` by one element, I feel like it's okay if we're okay with publishing single elements which are larger than the publish size.